presto-main/src/main/java/com/facebook/presto/operator/ExplainAnalyzeOperator.java

```diff
@@ -134,13 +136,26 @@ public Page getOutput()
         if (!finishing) {
             return null;
         }
-        outputConsumed = true;
+
         QueryInfo queryInfo = queryPerformanceFetcher.getQueryInfo(operatorContext.getDriverContext().getTaskId().getQueryId());
+
         // Skip the output stage, since that's the one that has the ExplainAnalyzeOperator itself
         checkState(queryInfo.getOutputStage().isPresent(), "Output stage is missing");
+
+        // Check if all stages, except the output stage, have the final StageInfo
+        boolean finalStageInfo = getAllStages(queryInfo.getOutputStage()).stream()
```

Comment:
I can add a new API `getFinalQueryInfo` which returns a future that this operator can wait on.